### PR TITLE
#184347101 ft-wishList

### LINKED ISSUE #41: A buyer should be able to add a product to wish list
### Why:
The backend needs to handle the addition of items to the buyer's wishlist, including storing the wishlist in the buyer's account and sending a confirmation message to the frontend.

### Acceptance criteria:

GIVEN a post request to add an item to the wishlist

### THEN:

- The item should be added to the buyer's wishlist
- The product details (name, price, image) should be retrieved from the database
- The updated wishlist should be stored in the buyer's account
- A confirmation message should be sent to the frontend to be displayed to the buyer

Dev notes:

- Validate that the buyer is logged in.
- Add the item to the buyer's wishlist.
- Retrieve the product details from the database.
- Store the updated wishlist in the buyer's account.
- Return a confirmation message to the frontend."

### PR DESCRIPTION
**Pull request by teerenzo**
### What does this PR do?

A buyer should be able to add a product to wish list

### Description of Task to be completed?

User will be able to add or remove product to wishList 

### How should this be manually tested?

using Swagger documentation

### Any background context you want to provide?

none

### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/184347101

### Questions:

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code is formatted correctly
- [x] My changes doesn't have any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by UwaweTera — DISMISSED** [review]

**Review by Ntarecedrick — DISMISSED** [review]

File: src/controllers/wishListController.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/* eslint-disable valid-jsdoc */
+/* eslint-disable require-jsdoc */
```
Comment (jimmymut) [inline src/controllers/wishListController.js]: These two lines can be one, /* eslint-disable valid-jsdoc, require-jsdoc */

File: src/database/models/productwishes.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/* eslint-disable valid-jsdoc */
+/* eslint-disable require-jsdoc */
```
Comment (jimmymut) [inline src/database/models/productwishes.js:2]: also here

File: src/middlewares/checkUserExist.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable no-else-return */
```
Comment (jimmymut) [inline src/middlewares/checkUserExist.js]: Does this comment needed?

**Review by Rutarenzi — DISMISSED** [review]

File: src/middlewares/checkUserExist.js (hunk 3 above)
Comment (jimmymut) [reply src/middlewares/checkUserExist.js]: if it is, then combine both comments in one!

**Review by jimmymut — DISMISSED** [review]

**Review by iradukundairenee — CHANGES_REQUESTED** [review]
1.please try looking at how you write your status code and  example on middleware/check productexist  file you use the status code 403 (forbidden) which is not good ,you can use 404 (notfound)
also you on wishListController on line 10 you have used 200 which is not Good there you can use 201 (created)
https://medium.com/rtkal/http-response-status-codes-b78f273ac56e

File: src/middlewares/checkUserExist.js (hunk 3 above)
Comment (teerenzo) [reply src/middlewares/checkUserExist.js]: yes I need it and how it is structured that is the best way for it to work

**Review by iradukundairenee — DISMISSED** [review]

**Review by UwaweTera — DISMISSED** [review]

**Review by Ntarecedrick — DISMISSED** [review]

File: src/database/migrations/20230216073036-create-product-wishes.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.createTable('ProductWishes', {
```
Comment (Eli250) [inline src/database/migrations/20230216073036-create-product-wishes.js:4]: Rename the model to `WishList` or `ProductWishList`

File: src/database/models/productwishes.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+const {
+  Model
+} = require('sequelize');
+
+module.exports = (sequelize, DataTypes) => {
+  class ProductWishes extends Model {
```
Comment (Eli250) [inline src/database/models/productwishes.js:9]: Rename class to `ProductWishList`

File: src/database/seeders/20230216134243-ProductWishes.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
+  async up (queryInterface, Sequelize) {
```
Comment (Eli250) [inline src/database/seeders/20230216134243-ProductWishes.js:2]: You have to provide at least 2 wishlist items here.

File: src/middlewares/checkUserExist.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable require-jsdoc */
+
+import { User } from "../database/models/index";
```
Comment (Eli250) [inline src/middlewares/checkUserExist.js:3]: I think we already have a middleware that checks this. If it is there, get rid of this file and use existing middleware.

File: src/routes/api/product.route.js [hunk 8]
```diff
@@ -22,4 +24,10 @@ route.get(
   ProductController.getAllItems
 );
 route.get("/:id", extractToken, ProductController.getSingleItem);
+route.get("/:id/product-wishes",
```
Comment (Eli250) [inline src/routes/api/product.route.js]: productWishList

**Review by Eli250 — CHANGES_REQUESTED** [review]

**Review by Ntarecedrick — APPROVED** [review]

**Review by Rutarenzi — APPROVED** [review]

**Review by NG-Solennel — APPROVED** [review]

**Review by Eli250 — APPROVED** [review]

**Pull request merged**